app/controllers/hearing-requirements/hearing-requirements-summary-sections.ts

```diff
@@ -630,3 +614,126 @@ function getInterpreterLanguageAnswer(language: InterpreterLanguageRefData): str
+    appellantInterpreterLanguageCategory += appellantInterpreterLanguageCategoryList.includes('signLanguageInterpreter') ? 'Sign language interpreter' : '';
+  }
+  interpreterRows.push(
+    getSummaryRow(visibleChangeLink,
+      i18n.common.cya.answerRowTitle,
+      [appellantInterpreterLanguageCategory],
+      paths.submitHearingRequirements.hearingInterpreterTypes
+    )
+  );
+
+  if (hearingRequirements.appellantInterpreterSpokenLanguage
+    && appellantInterpreterLanguageCategoryList
+    && appellantInterpreterLanguageCategoryList.includes('spokenLanguageInterpreter')) {
+    interpreterRows.push(
+      getSummaryRow(visibleChangeLink,
+        i18n.common.cya.questionRowTitle,
+        [i18n.pages.hearingRequirements.accessNeedsSection.interpreterSpokenLanguageSelection.title]
+      )
+    );
+
+    interpreterRows.push(
+      getSummaryRow(visibleChangeLink,
+        i18n.common.cya.answerRowTitle,
+        [getInterpreterLanguageAnswer(hearingRequirements.appellantInterpreterSpokenLanguage)],
+        paths.submitHearingRequirements.hearingInterpreterSpokenLanguageSelection
+      )
+    );
+  }
+
+  if (hearingRequirements.appellantInterpreterSignLanguage
+    && appellantInterpreterLanguageCategoryList
+    && appellantInterpreterLanguageCategoryList.includes('signLanguageInterpreter')) {
+    interpreterRows.push(
+      getSummaryRow(visibleChangeLink,
+        i18n.common.cya.questionRowTitle,
+        [i18n.pages.hearingRequirements.accessNeedsSection.interpreterSignLanguageSelection.title]
+      )
+    );
+
+    interpreterRows.push(
+      getSummaryRow(visibleChangeLink,
+        i18n.common.cya.answerRowTitle,
+        [getInterpreterLanguageAnswer(hearingRequirements.appellantInterpreterSignLanguage)],
+        paths.submitHearingRequirements.hearingInterpreterSignLanguageSelection
+      )
+    );
+  }
+}
+
+function buildWitnessesInterpretersummaryList(interpreterRows: SummaryRow[], visibleChangeLink: boolean, hearingRequirements: HearingRequirements, witnessComponent: WitnessComponent) {
+  interpreterRows.push(
+    addSummaryRow(
+      i18n.common.cya.questionRowTitle,
+      [i18n.pages.hearingRequirements.accessNeedsSection.interpreterTypePage.witnessTitle.replace('{witnessName}', witnessComponent.witnessFullName)]
+    )
+  );
+
+  let witnessInterpreterLanguageCategory: string = '';
+  if (witnessComponent.witnessInterpreterLanguageCategory && witnessComponent.witnessInterpreterLanguageCategory.length > 0) {
+    witnessInterpreterLanguageCategory += witnessComponent.witnessInterpreterLanguageCategory.includes('spokenLanguageInterpreter') ? 'Spoken language interpreter' : '';
+    witnessInterpreterLanguageCategory += (witnessComponent.witnessInterpreterLanguageCategory.length === 2) ? Delimiter.BREAK_LINE : '';
+    witnessInterpreterLanguageCategory += witnessComponent.witnessInterpreterLanguageCategory.includes('signLanguageInterpreter') ? 'Sign language interpreter' : '';
+  }
```

Comment:
The extracted helper function could be used here. Since this piece of code does exactly what the previous piece does.